src/Functions/readWkb.cpp

```diff
@@ -0,0 +1,315 @@
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeCustomGeo.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/geometryConverters.h>
+
+#include "Common/Exception.h"
+#include <Common/WKB.h>
+
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <variant>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+namespace
+{
+
+template <typename T>
+T convertGeometricObjectToBoostObject(const ArrowGeometricObject & object)
```

Comment:
Why aren't we reading into boost types directly? What's the point of using a different type just to convert it later?